src/Symfony/Component/VarDumper/Test/VarDumperTestTrait.php

```diff
@@ -31,11 +52,14 @@ public function assertDumpMatchesFormat($expected, $data, $filter = 0, $message
 
     protected function getDump($data, $key = null, $filter = 0)
     {
-        $flags = getenv('DUMP_LIGHT_ARRAY') ? CliDumper::DUMP_LIGHT_ARRAY : 0;
-        $flags |= getenv('DUMP_STRING_LENGTH') ? CliDumper::DUMP_STRING_LENGTH : 0;
-        $flags |= getenv('DUMP_COMMA_SEPARATOR') ? CliDumper::DUMP_COMMA_SEPARATOR : 0;
+        if (null === $flags = $this->varDumperTestTraitConfig['flags']) {
+            $flags = getenv('DUMP_LIGHT_ARRAY') ? CliDumper::DUMP_LIGHT_ARRAY : 0;
+            $flags |= getenv('DUMP_STRING_LENGTH') ? CliDumper::DUMP_STRING_LENGTH : 0;
+            $flags |= getenv('DUMP_COMMA_SEPARATOR') ? CliDumper::DUMP_COMMA_SEPARATOR : 0;
+        }
 
         $cloner = new VarCloner();
+        $cloner->addCasters($this->varDumperTestTraitConfig['casters']);
         $cloner->setMaxItems(-1);
```

Comment:
im still curious :) if we move this last entry (max_items=-1) to the config as well (e.g. setUpVarDumper), then we should be able to initialize the cloner using `@before` no? while you can still calll setUpVarDumper on demand.